azure/services/groups/spec.go

```diff
@@ -38,7 +39,7 @@ type GroupSpec struct {
 func (s *GroupSpec) ResourceRef() *asoresourcesv1.ResourceGroup {
 	return &asoresourcesv1.ResourceGroup{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: s.Name,
+			Name: azure.GetNormalizedKubernetesName(s.Name),
```

Comment:
I'd vote instead to create a new separate `AzureName` field in `GroupSpec` that gets set in scope/managedcontrolplane.go because I think that will make it easier to reason about the difference between it and `s.Name` throughout this file and reduces the chance that we inadvertently change the name of existing resource groups.